src/main/java/org/passay/NumberRangeRule.java

```diff
@@ -0,0 +1,135 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Rule for determining if a password contains any number within a defined range, lower inclusive, upper exclusive.
+ *
+ * @author  Middleware Services
+ */
+public class NumberRangeRule implements Rule
```

Comment:
I must admit I find this rule strange. I can't imagine a use case where this rule would be used for anything other than a sequence of decimal digits, e.g `new NumberRangeRule(0, 10)`. Doesn't that do exactly the same thing as `NumericalSequenceRule`? Of course you could restrict to a smaller range, but again I think the name and semantics of `NumericalSequenceRule` fit better. Why not improve the existing rule with range customization?